addons/web/static/src/views/fields/dynamic_placeholder_popover.js

```diff
@@ -28,8 +48,9 @@ export class DynamicPlaceholderPopover extends Component {
         }
         this.props.close();
     }
-    setPath(path) {
+    setPath(path, fieldInfo) {
         this.state.path = path;
+        this.state.fieldName = fieldInfo && fieldInfo.string;
```

Comment:
You can shorten this to
```suggestion
        this.state.fieldName = fieldInfo?.string;
```
[Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) lets you do the same. 
Unless you absolutely need the value to be false and a *nullish* value doesn't work in your flow.